pkg/controller/managedresource/add.go

```diff
@@ -55,12 +58,24 @@ type ControllerConfig struct {
 	SyncPeriod           time.Duration
 	ClassFilter          *filter.ClassFilter
 	AlwaysUpdate         bool
+	ClusterID            string
 
 	TargetClientConfig resourcemanagercmd.TargetClientConfig
 }
 
 // AddToManagerWithOptions adds the controller to a Manager with the given config.
 func AddToManagerWithOptions(mgr manager.Manager, conf ControllerConfig) error {
+	if conf.ClusterID == "<cluster>" || conf.ClusterID == "<default>" {
```

Comment:
> <cluster>: The cluster MUST have a cluster identity that is used.
<default>: If the cluster has a cluster identity it is used, otherwise a local scenario is assumed where the cluster is omitted from the origin
...

Do we need all this complexity? Can't we make this cluster ID optional and implement the following behaviour:

1. If `--cluster-id=some-value` was provided then use `some-value` as cluster ID.
2. If not/if it's empty then try to lookup the `cluster-identity` config map. If it exists then use the value from here, otherwise there is no cluster identity (we can log this, of course)

Reply:
The grm can be used outside the gardener also, therefore we should have a possibility to enforce the usage of a central cluster id. This is this `<cluster>` value.
On the other hand side it should be possible to enforce NO id or to use an Id if available.
There are the current cases_
  - anyid: use this one
  - empty: use none
  - `<cluster>`: enforce central cluster id
  - `<default>`: use central cluster id if possible

The defaulting case may potentially be omitted, but it does not really introduce complexity, so for the sake of completeness I would keep it.

Reply:
OK, fine then, I think I got it now. Thanks for the explanations. Can you please add a short `cluster_id.md` Markdown file to https://github.com/gardener/gardener-resource-manager/tree/master/docs/concepts and describe this parameter? I think it's important to keep this well documented, especially with the "special" values `<cluster>` and `<default>` (the PR will later be hard to find).

Reply:
I had put the docu to the existing md file